compel/src/lib/infect.c

```diff
@@ -447,6 +447,7 @@ static int parasite_run(pid_t pid, int cmd, unsigned long ip, void *stack,
 	k_rtsigset_t block;
 
 	ksigfillset(&block);
+	block.sig[0] = block.sig[0] & ~(1<<(SIGTRAP - 1));
```

Comment:
It is better to have a comment that explains why SIGTRAP isn't blocked.
We need to check that a process doesn't have a sigtrap signal in queues.

Reply:
Yes, this probably should be explained a little bit more... Why not-blocking SIGTRAP helps? PTRACE_SETSIGINFO seem to only change task_struct->blocked https://github.com/torvalds/linux/blob/8ac91e6c6033ebc12c5c1e4aa171b81a662bd70f/kernel/ptrace.c#L1117 But SigCgt is determined from the fact that actual signal handler was set https://github.com/torvalds/linux/blob/8ac91e6c6033ebc12c5c1e4aa171b81a662bd70f/fs/proc/array.c#L262. From the firs look they seem independent.... Sorry if I miss something obvious.

Reply:
this appeared to myself as a workaround ... for the existing bug that the caught sigset is not restored correctly as the checkpoint process was, as shown in this investigation:  https://github.com/checkpoint-restore/criu/issues/1429#issuecomment-815802412   and  https://github.com/checkpoint-restore/criu/issues/1429#issuecomment-815807630

this issue existed on zLinux too.  Although i didn't verify this on x86 linux, i expected it is there too.  There might be a better place to fix this, rather than in infect.c.  However, I am not familiar with the whole code enough to locate where it should be.

Reply:
@Snorch in the issue, I explained why we can't block SIGTRAP here and how it affects SigCgt.

@zl-wang Thank you for the patch. I understand that you are not familiar with the CRIU code, so I will fix the patch and merge it.